app/assets/javascripts/discourse/app/components/bulk-select-topics-dropdown.gjs

```diff
@@ -186,6 +202,23 @@ export default class BulkSelectTopicsDropdown extends Component {
     await this.dMenu.close();
 
     switch (actionId) {
+      case "dismiss-unread":
+        this.modal.show(DismissReadModal, {
+          model: {
+            title: "topics.bulk.dismiss_read_with_selected",
+            count: this.args.bulkSelectHelper.selected.length,
+            dismissRead: (dismissTopics) => this.dismissRead(dismissTopics),
+          },
+        });
+        break;
+      case "dismiss-new":
+        this.modal.show(DismissNew, {
+          model: {
+            selectedTopics: this.args.bulkSelectHelper.selected,
+            dismissCallback: (dismissTopics) => this.dismissRead(dismissTopics),
```

Comment:
It feels a bit odd that we're calling `dismissRead` from the `DismissNew` modal. It seems to work fine, per the system spec. Maybe we should rename the helper to `dismissPosts` and then pass the type (read/new) as an argument? This would help with the case when dismissing without selecting, currently that defaults to dismissing the `unread` posts if nothing is selected (not sure we expose this in the UI): 

```
Topic.bulkOperationByFilter("unread", operation, options, isTracked);
```

Reply:
I've added a not for me to look into this next week. 🙏